### ISSUE TITLE
[F#] Implement new Concept Exercise: overflows

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#856]
This issue describes how to implement the `overflows` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Memory Allocation in F#.

## Learning objectives

- Explain integral number overflows.
- Explain floating-point number overflows.
- Know how to use the `Checked` module to deal with overflows.

## Out of scope

- BigInteger

## Concepts

The Concepts this exercise unlocks are:

- `overflows`: explain integral number overflows; explain floating-point number overflows; know how to use the `Checked` module to deal with overflows.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `integral-numbers`: know about the integral number types.
- `floating-point-numbers`: know about the floating-point number types.
- `exceptions`: explain how an `OverflowException` is thrown when using checked arithmetic.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Arithmetic operators][arithmetic-operators]: describes the arithmetic operators and which are checked vs unchecked.
- [Checked context][checked-context]: describes how to do checked arithmetic in C# and F#.
- [Computerphile: How Gangnam Style broke YouTube][computerphile-gangnam-style]

### After

- [Arithmetic operators][arithmetic-operators]: describes the arithmetic operators and which are checked vs unchecked.
- [Checked context][checked-context]: describes how to do checked arithmetic in C# and F#.
- [Computerphile: How Gangnam Style broke YouTube][computerphile-gangnam-style]

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/concepts/overflows.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[computerphile-gangnam-style]: https://www.youtube.com/watch?v=vA0Rl6Ne5C8
[arithmetic-operators]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/symbol-and-operator-reference/arithmetic-operators
[checked-context]: https://theburningmonk.com/2012/01/checked-context-in-c-and-f/
